ibm/service/power/data_source_ibm_pi_volumes.go

```diff
@@ -0,0 +1,255 @@
+						},
+						Attr_ReplicationSites: {
+							Computed:    true,
+							Description: "List of replication sites for volume replication.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Type:        schema.TypeList,
+						},
+						Attr_ReplicationStatus: {
+							Computed:    true,
+							Description: "The replication status of the volume.",
+							Type:        schema.TypeString,
+						},
+						Attr_ReplicationType: {
+							Computed:    true,
+							Description: "The replication type of the volume, metro or global.",
+							Type:        schema.TypeString,
+						},
+						Attr_Shareable: {
+							Computed:    true,
+							Description: "Indicates if the volume is shareable between VMs.",
+							Type:        schema.TypeBool,
+						},
+						Attr_Size: {
+							Computed:    true,
+							Description: "The size of the volume in GB.",
+							Type:        schema.TypeInt,
+						},
+						Attr_State: {
+							Computed:    true,
+							Description: "The state of the volume.",
+							Type:        schema.TypeString,
+						},
+						Attr_UserTags: {
+							Computed:    true,
+							Description: "List of user tags attached to the resource.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Set:         schema.HashString,
+							Type:        schema.TypeSet,
+						},
+						Attr_VolumeName: {
+							Computed:    true,
+							Description: "The name of the volume.",
+							Type:        schema.TypeString,
+						},
+						Attr_VolumePool: {
+							Computed:    true,
+							Description: "Volume pool, name of storage pool where the volume is located.",
+							Type:        schema.TypeString,
+						},
+						Attr_WWN: {
+							Computed:    true,
+							Description: "The world wide name of the volume.",
+							Type:        schema.TypeString,
+						},
+					},
+				},
+				Type: schema.TypeList,
+			},
+		},
+	}
+}
+
+func dataSourceIBMPIVolumesRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, fmt.Sprintf("IBMPISession failed: %s", err.Error()), "(Data) ibm_pi_volumes", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	volumeC := instance.NewIBMPIVolumeClient(ctx, sess, cloudInstanceID)
+	volumeData, err := volumeC.GetAll()
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, fmt.Sprintf("GetAll failed: %s", err.Error()), "(Data) ibm_pi_volumes", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	var clientgenU, _ = uuid.GenerateUUID()
+	d.SetId(clientgenU)
+	d.Set(Attr_Volumes, flattenVolumes(volumeData.Volumes, meta))
+	return nil
+}
+
+func flattenVolumes(list []*models.VolumeReference, meta any) []map[string]any {
+	log.Printf("Calling the flattenVolumeGroups call with list %d", len(list))
```

Comment:
Any reason why we are logging this?

Reply:
No I think some of the other functions were doing it so I just copied them.